smarts/core/smarts.py

```diff
@@ -651,6 +651,11 @@ def providers(self):
         # TODO: It's inconsistent that pybullet is not here
         return self._providers
 
+    def get_provider_by_type(self, requested_type):
+        for provider in self._providers:
+            if isinstance(provider, requested_type):
+                return provider
```

Comment:
This is good for now, we can improve it can be optimised later.

Reply:
If it is a small optimization we could do it now. I found this interesting way to do it. would this make it more optimized? 
```
     def get_provider_by_type(self, requested_type): 
         return next((p for p in self._providers if isinstance(p, requested_type) None)
```